### PR TITLE
Migrate .bss for fault drawer and .data for item00

### PR DESCRIPTION
**Pull request by louist103**
Also fixed up item00 to fit out current standards.

### CODE REVIEW

**Review by Zelllll — APPROVED** [review]
Great job, looks good to me!

File: src/code/z_en_item00.c [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #include "global.h"
 #include "overlays/actors/ovl_En_Elf/z_en_elf.h"
 #include "overlays/effects/ovl_Effect_Ss_Dead_Sound/z_eff_ss_dead_sound.h"
-#include "objects/gameplay_keep/gameplay_keep.h"
```
Comment (fig02) [inline src/code/z_en_item00.c:4]: theres definitely things from gameplay keep in here. how is this compiling?

Comment (louist103) [reply src/code/z_en_item00.c:4]: The gameplay keep header is in the .data.c file

Comment (fig02) [reply src/code/z_en_item00.c:4]: ah. as a seperate point from that, I was going to say that imo a data.c isnt really needed here

File: src/code/z_en_item00.c [hunk 2]
```diff
@@ -370,32 +653,32 @@ void func_8001E304(EnItem00* this, GlobalContext* globalCtx) {
...
     if (this->actor.velocity.y <= 2.0f) {
-        var1 = (u16)this->actor.shape.rot.z + 10000;
-        if (var1 < 65535) {
-            this->actor.shape.rot.z += 10000;
+        var1 = (u16)this->actor.shape.rot.z + 0x2710;
+        if (var1 < 0xFFFF) {
+            this->actor.shape.rot.z += 0x2710;
         } else {
-            this->actor.shape.rot.z = -1;
+            this->actor.shape.rot.z = -0x0001;
```
[the remark this answers is not in the document]
Comment (fig02) [reply src/code/z_en_item00.c:668]: yea if an angle looks good/rounded in decimal, it can stay as decimal. this goes for small numbers as well

**Comment by fig02** [comment]
after these angle changes are reverted this should be good to go

**Comment by fig02** [comment]
this doesnt match for some reason. theres no build errors so something is causing an actual non-match

**Comment by louist103** [comment]
> 
> 
> this doesnt match for some reason. theres no build errors so something is causing an actual non-match

it matches locally tho.  Ill see whats going on.

**Review by fig02 — APPROVED** [review]

File: src/code/z_en_item00.c [hunk 3]
```diff
@@ -421,7 +702,7 @@ void func_8001E5C8(EnItem00* this, GlobalContext* globalCtx) {
     this->actor.world.pos = player->actor.world.pos;
 
     if (this->actor.params <= ITEM00_RUPEE_RED) {
-        this->actor.shape.rot.y += 960;
+        this->actor.shape.rot.y += 960  ;
```
Comment (Roman971) [inline src/code/z_en_item00.c:705]:
```suggestion
        this->actor.shape.rot.y += 960;
```

File: src/code/z_en_item00.c [hunk 4]
```diff
@@ -638,7 +919,7 @@ void EnItem00_Update(Actor* thisx, GlobalContext* globalCtx) {
 
     this->unk_15A = 15;
     this->unk_154 = 35;
-    this->actor.shape.rot.z = 0;
+    this->actor.shape.rot.z = 0x0000;
```
Comment (Roman971) [inline src/code/z_en_item00.c:922]:
```suggestion
    this->actor.shape.rot.z = 0;
```

File: src/code/z_en_item00.c [hunk 5]
```diff
@@ -29,19 +29,300 @@ const ActorInit En_Item00_InitVars = {
...
+};
 
-// Internal Actor Functions
+static Color_RGBA8 sEffectPrimColor = { 255, 255, 127, 0 };
+
+static Color_RGBA8 sEffectEnvColor = { 255, 255, 255, 0 };
+
+static Vec3f sEffectPos = { 0.0f, 0.1f, 0.0f };
+
+static Vec3f sEffectAcell = { 0.0f, 0.01f, 0.0f };
```
Comment (Roman971) [inline src/code/z_en_item00.c:62]:
```suggestion
static Vec3f sEffectAccel = { 0.0f, 0.01f, 0.0f };
```

**Review by Roman971 — COMMENTED** [review]
Just a few things then it looks good to me

**Review by Roman971 — APPROVED** [review]

**Pull request merged**